### PR TITLE
Fix/cta text required

### PR DESCRIPTION
**Pull request by William-Blackie**
This fixes the implied functionality of using the CTABlock model where required=False is now respected.

Changes: 
- Template checks for the value of the title as value.cta will return True regardless.
- Set the CTABlock text to required=False and validate when it's conditionally required.

### CODE REVIEW

File: wagtailio/core/blocks.py [hunk 1]
```diff
@@ -196,24 +196,17 @@ def url(self):
 
 
 class CTALinkMixin(blocks.StructBlock):
-    def is_link_required(self):
```
Comment (William-Blackie) [inline wagtailio/core/blocks.py:199]: We actually have the required value passed to Streamfield, we can use that instead.

File: wagtailio/core/blocks.py [hunk 2]
```diff
@@ -196,24 +196,17 @@ def url(self):
...
         errors = {}
         struct_value = super().clean(value)
-
-        if self.is_link_required():
-            if not value.get("cta_page") and not value.get("cta_url"):
-                error = ErrorList(
-                    [ValidationError("You must specify CTA page or CTA URL.")]
-                )
-                errors["cta_url"] = errors["cta_page"] = error
+        if self.required and not value.get("cta_page") and not value.get("cta_url"):
```
Comment (William-Blackie) [inline wagtailio/core/blocks.py:205]: This is a simplification of the above.

File: wagtailio/core/blocks.py [hunk 3]
```diff
@@ -225,9 +218,12 @@ def clean(self, value):
             )
             errors["cta_url"] = errors["cta_page"] = error
 
+        if not value.get("text") and (value.get("cta_page") or value.get("cta_url")):
```
Comment (William-Blackie) [inline wagtailio/core/blocks.py:221]: Require text, if someones added a link.

**Review by zerolab — APPROVED** [review]
LGTM.

Had a small nitpick on the unnecessary `error` variable assignment for the empty text field check, but it is just that, so let's get this moving :shipit:

**Pull request merged**